azurerm/internal/acceptance/data.go

```diff
@@ -82,3 +83,30 @@ func BuildTestData(t *testing.T, resourceType string, resourceLabel string) Test
 
 	return testData
 }
+
+func (td *TestData) RandomIntOfLength(len int) int {
+	// len should not be
+	//  - greater then 18, longest a int can represent
+	//  - less then 8, as that gives us YYMMDDRR
+	if 8 > len || len > 18 {
+		panic(fmt.Sprintf("Invalid Test: RandomIntOfLength: len is not between 6 or 18 inclusive"))
+	}
+
+	r := td.RandomInteger % 100
+
+	// 18 - just return the int
+	if len >= 18 {
+		return td.RandomInteger
+	}
+
+	// 16-17 just strip off the last 1-2 digits
+	if len >= 16 {
+		return td.RandomInteger / int(math.Pow10(18-len))
```

Comment:
I think this could be simplified to:

```suggestion
		return td.RandomInteger[0:15]
```

Reply:
results in a compilation error as its a int not a string